libraries/AP_Scripting/examples/plane-sweep.lua

```diff
@@ -0,0 +1,129 @@
+        -- choosing control actuator, 900 for elevator, 1100 for aileron, 1300 for rudder, 1500 for throttle 
+        if SWEEP_choice_pwm < 1000 then
+            SWEEP_FUCNTION = K_ELEVATOR
+            if SWEEP_srv_trim == -100 then
+                SWEEP_srv_trim = pre_SWEEP_elevator 
+            end
+        elseif SWEEP_choice_pwm > 1000 and SWEEP_choice_pwm < 1200 then
+            SWEEP_FUCNTION = K_AILERON
+            if SWEEP_srv_trim == -100 then
+                SWEEP_srv_trim = pre_SWEEP_aileron 
+            end
+        elseif SWEEP_choice_pwm > 1200 and SWEEP_choice_pwm < 1400 then
+            SWEEP_FUCNTION = K_RUDDER
+            if SWEEP_srv_trim == -100 then
+                SWEEP_srv_trim = pre_SWEEP_rudder 
+            end
+        elseif SWEEP_choice_pwm > 1400 and SWEEP_choice_pwm < 1600 then
+            SWEEP_FUCNTION = K_THROTTLE
+            if SWEEP_srv_trim == -100 then
+                SWEEP_srv_trim = pre_SWEEP_throttle
+            end
+        else
+            SWEEP_FUCNTION = K_ELEVATOR
+            if SWEEP_srv_trim == -100 then
+                SWEEP_srv_trim = pre_SWEEP_elevator 
+            end
+        end
+        local SWEEP_srv_chan = SRV_Channels:find_channel(SWEEP_FUCNTION)
+        local SWEEP_srv_min = param:get("SERVO" .. SWEEP_srv_chan + 1 .. "_MIN")
+        local SWEEP_srv_max = param:get("SERVO" .. SWEEP_srv_chan + 1 .. "_MAX")
+        local SWEEP_srv_trim = param:get("SERVO" .. SWEEP_srv_chan + 1 .. "_TRIM")
+        if start_time == -1 then
+            start_time = now
+            gcs:send_text(6, "STARTING SWEEP " .. SWEEP_srv_chan)
+            retry_set_mode(MODE_MANUAL)
+        end
+        if now < (start_time + (SWEEP_TIME)) then
+        --     gcs:send_text(6, "SWEEP up" .. SWEEP_srv_chan)
+        --     up = SWEEP_srv_trim + math.floor((SWEEP_srv_max - SWEEP_srv_trim) * (SWEEP_MAGNITUDE / 45))
+        --     SRV_Channels:set_output_pwm_chan_timeout(SWEEP_srv_chan, up, SWEEP_TIME / 2 + 100)
+        --     gcs:send_text(6, "pwm" .. tostring(SRV_Channels:get_output_pwm(SWEEP_FUCNTION)))
+        -- elseif now < (start_time + SWEEP_TIME) and now > (start_time + (SWEEP_TIME / 2))then
+        --     gcs:send_text(6, "SWEEP down" .. SWEEP_srv_chan)
+        --     down = SWEEP_srv_trim - math.floor((SWEEP_srv_trim - SWEEP_srv_min) * (SWEEP_MAGNITUDE / 45))
+        --     SRV_Channels:set_output_pwm_chan_timeout(SWEEP_srv_chan, down, SWEEP_TIME / 2 + 100)
+        --     gcs:send_text(6, "pwm" .. tostring(SRV_Channels:get_output_pwm(SWEEP_FUCNTION)))
+            gcs:send_text(6, "in sweep loop" .. SWEEP_srv_chan)
+            t_i = tonumber(tostring(now - start_time))
+            amplitude_sent = SWEEP_MAGNITUDE*math.sin(tonumber(2*math.pi*(start_freq * t_i + (k / 2) * t_i*t_i)))
+            amplitude_sent = math.floor(amplitude_sent)
+            SRV_Channels:set_output_pwm_chan_timeout(SWEEP_srv_chan,SWEEP_srv_trim + amplitude_sent,40)
```

Comment:
Your `callback_time` is 100ms, but you are capturing the output for 40ms.
Won't the flight controller revert to normal output levels for the remaining 60ms?